Fix failing mute till test

ibrkhalil opened the pull request:
fixes #16452 

status: ready

ibrkhalil committed:
Fix failing test

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :x: | 7c9b4783 | [#1](https://ci.status.im/job/status-mobile/job/prs/job/tests/job/PR-16453/1/) | 2023-06-30 13:38:06 | ~5 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-mobile/job/prs/job/tests/job/PR-16453/1/consoleText) |

ibrkhalil committed:
Lint

ilmotta commented on src/utils/datetime_test.cljs:
```diff
@@ -190,76 +193,75 @@
-      (testing "Until the user turns it back on"
-        (is (= "you turn it back on" (datetime/format-mute-till datetime/go-default-time))))))
+(deftest format-mute-till-test
+    (with-redefs [t/now (constantly (time-coerce/from-long mock-current-time-epoch))]
```
@ibrkhalil, I think the proper way is to redef the dynamic var `*ms-fn*` and not the core functions exposed by cljs-time. As we can see, this var is used throughout the cljs-time code https://github.com/andrewmcveigh/cljs-time/blob/master/src/cljs_time/core.cljs#L289. This is also how other tests in `datetime-test` do.

The dynamic var expects a function that returns a number in milliseconds, so you can skip `from-long` and do `(with-redefs [t/*ms-fn* (constantly mock-current-time-epoch)])`

J-Son89 commented on src/utils/datetime_test.cljs:
```diff
@@ -190,76 +193,75 @@
-         will-unmute-in-six-days   (write-date mock-in-six-days)))
-      (testing "Until the user turns it back on"
-        (is (= "you turn it back on" (datetime/format-mute-till datetime/go-default-time))))))
+(deftest format-mute-till-test
```
imo it seems like `format-mute-till` is quite application specific and this code is in a general `datetime` utilities. Is there a better location to have this code? 
cc @flexsurfer

ibrkhalil replied:
I think it'd be nicer if we have a muting_tests.cljs which has the muting specfic tests.
But for this case, It's a single test and it interacts with time functions so I think it's okish to have it here.

J-Son89 replied:
I was referring to the functionality not to the test itself. It seems like this functionality is misplaced and ofc the test should sit alongside it 👍

J-Son89 replied:
e.g src/contexts/common/muting seems more applicable imo

ibrkhalil replied:
Apologies for long response, Had this task on hold for a bit
Addressed here 351b87d9e335008028ad7382629f304646a4185e

status-im-auto commented:
## 85% of end-end tests have passed
```
Total executed tests: 33
Failed tests: 5
Passed tests: 28
```
```
IDs of failed tests: 702733,702732,702894,702855,702731 
```
<h3>Failed tests (5)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=develop&PR_ID=16453&TR_CASE_IDS=702733,702732,702894,702855,702731">Rerun failed tests</a></li><br/><h4>Class TestGroupChatMultipleDeviceMergedNewUI:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3183398">test_group_chat_pin_messages</a>, id: 702732 </b></td></tr><tr><td>

```
Test is not run, e2e blocker  
```

<b>[[reason: [NOTRUN] Pin feature is in development]]</b><br/><br/></td></tr></tbody></table><h4>Class TestOneToOneChatMultipleSharedDevicesNewUi:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3183399">test_1_1_chat_text_message_delete_push_disappear</a>, id: 702733 </b></td></tr><tr><td><p><blockquote><div>Device 2: Find `Text` by `xpath`: `//*[starts-with(@text,'smth I should edit')]/ancestor::android.view.ViewGroup[@content-desc='chat-item']//*[@content-desc='message-status']/android.widget.TextView` </div><div>Device 2: `Text` is `Sent` </div></blockquote></p>

```
critical/chats/test_1_1_public_chats.py:1219: in test_1_1_chat_text_message_delete_push_disappear
    self.chat_2.chat_element_by_text(message_after_edit_1_1).wait_for_status_to_be("Delivered")
../views/chat_view.py:229: in wait_for_status_to_be
    raise TimeoutException("Message status was not changed to %s, it's %s" % (expected_status, current_status))
 Message status was not changed to Delivered, it's Sent
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/619b79ba229b4fbfa251e12d111bd08d?auth=43ac7935c4e1274f7a82bf158da40db6#309">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/fa1a361fad8d4bfa8f0a7a2838ac7a8a?auth=54012be25fc0ada837a355e8025a1e0d#274">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3183419">test_1_1_chat_edit_message</a>, id: 702855 </b></td></tr><tr><td><p><blockquote><div>Device 2: Find `Text` by `xpath`: `//*[starts-with(@text,'Message before edit 1-1')]/ancestor::android.view.ViewGroup[@content-desc='chat-item']//*[@content-desc='message-status']/android.widget.TextView` </div><div>Device 2: `Text` is `Sent` </div></blockquote></p>

```
critical/chats/test_1_1_public_chats.py:1192: in test_1_1_chat_edit_message
    self.chat_2.chat_element_by_text(message_before_edit_1_1).wait_for_status_to_be("Delivered")
../views/chat_view.py:229: in wait_for_status_to_be
    raise TimeoutException("Message status was not changed to %s, it's %s" % (expected_status, current_status))
 Message status was not changed to Delivered, it's Sent
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/619b79ba229b4fbfa251e12d111bd08d?auth=43ac7935c4e1274f7a82bf158da40db6#295">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/fa1a361fad8d4bfa8f0a7a2838ac7a8a?auth=54012be25fc0ada837a355e8025a1e0d#218">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3183397">test_1_1_chat_pin_messages</a>, id: 702731 </b></td></tr><tr><td>

```
Test is not run, e2e blocker  
```

<b>[[reason: [NOTRUN] Pin feature is in development]]</b><br/><br/></td></tr></tbody></table><h4>Class TestCommunityMultipleDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3183421">test_community_contact_block_unblock_offline</a>, id: 702894 </b></td></tr><tr><td><p><blockquote><div>Device 1: Tap on found: Button </div><div>Device 1: Looking for a message by text: Hurray! unblocked </div></blockquote></p>

```
critical/test_public_chat_browsing.py:780: in test_community_contact_block_unblock_offline
    self.errors.verify_no_errors()
base_test_case.py:182: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Hurray! unblocked was not received in public chat after user unblock!
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/4ccd3414e23b4ce58f1eafe39be2cbbf?auth=8ac59601fdd58e374d4d160e9e076672#326">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/efff8cdb05d34b84b7b0d3e0cfd48fc1?auth=e042008633d912c12b7b04858c2d3393#266">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p></ul></p></td></tr></tbody></table></details><h3>Passed tests (28)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3183427">test_restore_multiaccount_with_waku_backup_remove_switch</a>, id: 703133 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/5dcc85df401a4104bac10898676396f2?auth=9fc2da9e5266430bf8e27c7d220d154e#117">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3183400">test_community_copy_and_paste_message_in_chat_input</a>, id: 702742 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/5dcc85df401a4104bac10898676396f2?auth=9fc2da9e5266430bf8e27c7d220d154e#92">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3183416">test_community_navigate_to_channel_when_relaunch</a>, id: 702846 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/5dcc85df401a4104bac10898676396f2?auth=9fc2da9e5266430bf8e27c7d220d154e#74">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityMultipleDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3183428">test_community_several_images_send_reply</a>, id: 703194 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/4ccd3414e23b4ce58f1eafe39be2cbbf?auth=8ac59601fdd58e374d4d160e9e076672#228">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/efff8cdb05d34b84b7b0d3e0cfd48fc1?auth=e042008633d912c12b7b04858c2d3393#170">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3183420">test_community_one_image_send_reply</a>, id: 702859 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/4ccd3414e23b4ce58f1eafe39be2cbbf?auth=8ac59601fdd58e374d4d160e9e076672#205">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/efff8cdb05d34b84b7b0d3e0cfd48fc1?auth=e042008633d912c12b7b04858c2d3393#149">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3183411">test_community_emoji_send_copy_paste_reply</a>, id: 702840 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/4ccd3414e23b4ce58f1eafe39be2cbbf?auth=8ac59601fdd58e374d4d160e9e076672#249">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/efff8cdb05d34b84b7b0d3e0cfd48fc1?auth=e042008633d912c12b7b04858c2d3393#208">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3183426">test_community_mark_all_messages_as_read</a>, id: 703086 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/4ccd3414e23b4ce58f1eafe39be2cbbf?auth=8ac59601fdd58e374d4d160e9e076672#454">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/efff8cdb05d34b84b7b0d3e0cfd48fc1?auth=e042008633d912c12b7b04858c2d3393#289">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3183405">test_community_mentions_push_notification</a>, id: 702786 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/4ccd3414e23b4ce58f1eafe39be2cbbf?auth=8ac59601fdd58e374d4d160e9e076672#476">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/efff8cdb05d34b84b7b0d3e0cfd48fc1?auth=e042008633d912c12b7b04858c2d3393#299">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>6. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3183410">test_community_message_delete</a>, id: 702839 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/4ccd3414e23b4ce58f1eafe39be2cbbf?auth=8ac59601fdd58e374d4d160e9e076672#184">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/efff8cdb05d34b84b7b0d3e0cfd48fc1?auth=e042008633d912c12b7b04858c2d3393#120">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>7. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3183409">test_community_message_send_check_timestamps_sender_username</a>, id: 702838 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/4ccd3414e23b4ce58f1eafe39be2cbbf?auth=8ac59601fdd58e374d4d160e9e076672#144">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/efff8cdb05d34b84b7b0d3e0cfd48fc1?auth=e042008633d912c12b7b04858c2d3393#103">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>8. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3183414">test_community_links_with_previews_github_youtube_twitter_gif_send_enable</a>, id: 702844 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/4ccd3414e23b4ce58f1eafe39be2cbbf?auth=8ac59601fdd58e374d4d160e9e076672#270">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/efff8cdb05d34b84b7b0d3e0cfd48fc1?auth=e042008633d912c12b7b04858c2d3393#223">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>9. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3183413">test_community_message_edit</a>, id: 702843 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/4ccd3414e23b4ce58f1eafe39be2cbbf?auth=8ac59601fdd58e374d4d160e9e076672#162">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/efff8cdb05d34b84b7b0d3e0cfd48fc1?auth=e042008633d912c12b7b04858c2d3393#115">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>10. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3183415">test_community_leave</a>, id: 702845 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/4ccd3414e23b4ce58f1eafe39be2cbbf?auth=8ac59601fdd58e374d4d160e9e076672#489">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/efff8cdb05d34b84b7b0d3e0cfd48fc1?auth=e042008633d912c12b7b04858c2d3393#322">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>11. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3183412">test_community_unread_messages_badge</a>, id: 702841 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/4ccd3414e23b4ce58f1eafe39be2cbbf?auth=8ac59601fdd58e374d4d160e9e076672#301">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/efff8cdb05d34b84b7b0d3e0cfd48fc1?auth=e042008633d912c12b7b04858c2d3393#258">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestActivityCenterContactRequestMultipleDevicePR:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3183418">test_activity_center_contact_request_accept_swipe_mark_all_as_read</a>, id: 702851 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d1fc4c88a0374ea6b2780888b5709970?auth=4e4b162d7e9cd5222325ad82906c93fb#89">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/b6596d9cb1a241ba8e9ffee8740404b6?auth=8654b1347f51fe301c4fd55c564f4e18#92">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3183417">test_activity_center_contact_request_decline</a>, id: 702850 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d1fc4c88a0374ea6b2780888b5709970?auth=4e4b162d7e9cd5222325ad82906c93fb#56">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/b6596d9cb1a241ba8e9ffee8740404b6?auth=8654b1347f51fe301c4fd55c564f4e18#55">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestGroupChatMultipleDeviceMergedNewUI:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3183406">test_group_chat_join_send_text_messages_push</a>, id: 702807 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/11856bb1a83240dead25732870ef175f?auth=cc584fe25cd79527faaa6149ee6e4440#121">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/74c39004755449bb90b4355c0d1f90e0?auth=99a8132fe1819d97d7903e94996d4e97#67">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/950ff964cba243f797413cb4fddb1483?auth=5981aad472e3bc40c8950bcd6d124c24#67">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3183407">test_group_chat_offline_pn</a>, id: 702808 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/11856bb1a83240dead25732870ef175f?auth=cc584fe25cd79527faaa6149ee6e4440#135">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/74c39004755449bb90b4355c0d1f90e0?auth=99a8132fe1819d97d7903e94996d4e97#92">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/950ff964cba243f797413cb4fddb1483?auth=5981aad472e3bc40c8950bcd6d124c24#72">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestOneToOneChatMultipleSharedDevicesNewUi:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3183402">test_1_1_chat_emoji_send_reply_and_open_link</a>, id: 702782 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/619b79ba229b4fbfa251e12d111bd08d?auth=43ac7935c4e1274f7a82bf158da40db6#129">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/fa1a361fad8d4bfa8f0a7a2838ac7a8a?auth=54012be25fc0ada837a355e8025a1e0d#106">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3183404">test_1_1_chat_delete_via_long_press_relogin</a>, id: 702784 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/619b79ba229b4fbfa251e12d111bd08d?auth=43ac7935c4e1274f7a82bf158da40db6#349">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/fa1a361fad8d4bfa8f0a7a2838ac7a8a?auth=54012be25fc0ada837a355e8025a1e0d#369">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3183408">test_1_1_chat_push_emoji</a>, id: 702813 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/619b79ba229b4fbfa251e12d111bd08d?auth=43ac7935c4e1274f7a82bf158da40db6#252">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/fa1a361fad8d4bfa8f0a7a2838ac7a8a?auth=54012be25fc0ada837a355e8025a1e0d#191">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3183401">test_1_1_chat_non_latin_messages_stack_update_profile_photo</a>, id: 702745 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/619b79ba229b4fbfa251e12d111bd08d?auth=43ac7935c4e1274f7a82bf158da40db6#191">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/fa1a361fad8d4bfa8f0a7a2838ac7a8a?auth=54012be25fc0ada837a355e8025a1e0d#147">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3183403">test_1_1_chat_is_shown_message_sent_delivered_from_offline</a>, id: 702783 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/619b79ba229b4fbfa251e12d111bd08d?auth=43ac7935c4e1274f7a82bf158da40db6#314">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/fa1a361fad8d4bfa8f0a7a2838ac7a8a?auth=54012be25fc0ada837a355e8025a1e0d#321">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>6. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3183396">test_1_1_chat_message_reaction</a>, id: 702730 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/619b79ba229b4fbfa251e12d111bd08d?auth=43ac7935c4e1274f7a82bf158da40db6#77">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/fa1a361fad8d4bfa8f0a7a2838ac7a8a?auth=54012be25fc0ada837a355e8025a1e0d#73">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestActivityMultipleDevicePR:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3183424">test_activity_center_mentions</a>, id: 702957 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/ffe2b70cf2a344f08ba116e95bea11bd?auth=7a0e5fa6655700abf8f20a351bf9102a#276">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9ff0be3794ce495b8c4d4778a5fa3ef5?auth=d2657bff2bebfeff35a01bdcfd9924b3#123">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3183422">test_navigation_jump_to</a>, id: 702936 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/ffe2b70cf2a344f08ba116e95bea11bd?auth=7a0e5fa6655700abf8f20a351bf9102a#143">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9ff0be3794ce495b8c4d4778a5fa3ef5?auth=d2657bff2bebfeff35a01bdcfd9924b3#104">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3183423">test_activity_center_reply_read_unread_delete_filter_swipe</a>, id: 702947 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/ffe2b70cf2a344f08ba116e95bea11bd?auth=7a0e5fa6655700abf8f20a351bf9102a#164">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9ff0be3794ce495b8c4d4778a5fa3ef5?auth=d2657bff2bebfeff35a01bdcfd9924b3#107">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3183425">test_activity_center_admin_notification_accept_swipe</a>, id: 702958 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/ffe2b70cf2a344f08ba116e95bea11bd?auth=7a0e5fa6655700abf8f20a351bf9102a#312">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9ff0be3794ce495b8c4d4778a5fa3ef5?auth=d2657bff2bebfeff35a01bdcfd9924b3#141">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details>

ilmotta commented on src/utils/datetime_test.cljs:
```diff
@@ -190,76 +193,75 @@
-          will-unmute-in-8-hours    (remove-msecs (time-coerce/to-string mock-in-8-hour))
-          will-unmute-in-15-mins    (remove-msecs (time-coerce/to-string mock-in-15-minutes))
-          will-unmute-in-two-days   (remove-msecs (time-coerce/to-string mock-in-two-days))
-          will-unmute-tomorrow      (remove-msecs (time-coerce/to-string mock-tomorrow))
```
@ibrkhalil, my suggestion here is not to be necessarily applied in this PR. Unit tests should be easy to read and the expected values should be often hardcoded.

If we set the current cljs-time to a fixed timestamp, then I believe all expected values can be transformed to simple strings and devs will instantly know what this function does and what it produces as a result of any input -> output. I can tell you, I had to start the REPL to know what `format-mute-till` produces.

ibrkhalil replied:
Understood, I don't think it's worth it to compute the expected values if we already know them, And like you just said it adds extra complexity in understanding what the test does.
I am back Monday, Will create a follow up for the refactor.

ibrkhalil replied:
https://github.com/status-im/status-mobile/issues/16458

ibrkhalil committed:
Comments

churik commented:
@ibrkhalil 
no manual testing is required, so you can merge it

ibrkhalil committed:
Folder structure

ibrkhalil committed:
Finalize

ibrkhalil committed:
Merge branch 'develop' into fix-failing-mute-till-test

ibrkhalil committed:
Lint

ibrkhalil committed:
git pushMerge branch
'fix-failing-mute-till-test' of
https://www.github.com/status-im/status-mobile
into fix-failing-mute-till-test
